src/TesApi.Tests/BatchSchedulerTests.cs

```diff
@@ -30,6 +30,68 @@ public class BatchSchedulerTests
         private static readonly Regex downloadFilesBlobxferRegex = new Regex(@"path='([^']*)' && url='([^']*)' && blobxfer download");
         private static readonly Regex downloadFilesWgetRegex = new Regex(@"path='([^']*)' && url='([^']*)' && mkdir .* wget");
 
+        [TestCategory("TES 1.1")]
+        [TestMethod]
+        public async Task BackendParametersVmSizeShallOverrideVmSelection()
+        {
+            // "vmsize" is not case sensitive
+            // If vmsize is specified, (numberofcores, memoryingb, resourcedisksizeingb) are ignored
+
+            var azureProxyReturnValues = AzureProxyReturnValues.Defaults;
+
+            azureProxyReturnValues.VmSizesAndPrices = new List<VirtualMachineInfo> {
+                new VirtualMachineInfo { VmSize = "VmSize1", LowPriority = true, NumberOfCores = 1, MemoryInGB = 4, ResourceDiskSizeInGB = 20, PricePerHour = 1 },
+                new VirtualMachineInfo { VmSize = "VmSize2", LowPriority = true, NumberOfCores = 2, MemoryInGB = 8, ResourceDiskSizeInGB = 40, PricePerHour = 2 }};
+
+            var backendParameters = new Dictionary<string, string>();
```

Comment:
To reduce repetitive lines `backendParameters = new Dictionary<string, string>();`
you can declare the variable once at the beginning: `Dictionary<string, string> backendParameters;`
and then subsequently set it: `backendParameters = new() {{ "vm_size", "VmSize1" }};`
Or just pass it directly: `state = await GetNewTesTaskStateAsync(new TesResources { Preemptible = true, BackendParameters = new() {{ "vm_size", "VmSize1" }} }, azureProxyReturnValues);`